repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts

```diff
@@ -490,9 +491,57 @@ export const combineFareZones = (outbound: FareZone[], inbound: FareZone[]): Far
 export const getLinesElement = (
     ticket: PointToPointTicket | PointToPointPeriodTicket,
     lineName: string,
+    isReturnAndHasAdditionalService: boolean,
 ): NetexObject => {
     const typeOfPointToPoint = ticket.type;
 
+    if (!isReturnAndHasAdditionalService) {
+        return {
+            version: '1.0',
+            id: `Tariff@${typeOfPointToPoint}@lines`,
+            Name: { $t: `O/D pairs for ${lineName}` },
+            TypeOfFareStructureElementRef: {
+                versionRef: 'fxc:v1.0',
+                ref: 'fxc:access',
+            },
+            distanceMatrixElements: {
+                DistanceMatrixElement: isReturnTicket(ticket)
+                    ? getDistanceMatrixElements(combineFareZones(ticket.outboundFareZones, ticket.inboundFareZones))
+                    : getDistanceMatrixElements(ticket.fareZones),
+            },
+            GenericParameterAssignment: {
+                version: '1.0',
+                order: '01',
+                id: `Tariff@${typeOfPointToPoint}@lines`,
+                TypeOfAccessRightAssignmentRef: {
+                    version: 'fxc:v1.0',
+                    ref: 'fxc:can_access',
+                },
+                ValidityParameterAssignmentType: { $t: 'EQ' },
+                validityParameters: {
+                    LineRef: {
+                        version: '1.0',
+                        ref: lineName,
+                    },
+                },
+            },
+        };
+    }
+
+    const lineRefs = [
+        {
+            version: '1.0',
+            ref: lineName,
+        },
+        {
+            version: '1.0',
+            ref:
+                isReturnTicket(ticket) && ticket.additionalServices && ticket.additionalServices.length > 0
+                    ? ticket.additionalServices[0]?.lineName
```

Comment:
Do we need the ? before [0] here? And I think we shouldnt have the situation where we can create an empty ref.

Reply:
it was taken from the code committed earlier ... let me remove it

Reply:
resolved in the file